src/StochasticStyles/spawning.jl

```diff
@@ -171,14 +156,24 @@ struct Exact{T} <: SpawningStrategy
     Exact(threshold::T=0.0) where {T} = new{T}(threshold)
 end
 
-@inline function spawn!(s::Exact, w, offdiags::AbstractVector, add, val, boost=1)
+"""
+    spawn!(s::SpawningStrategy, w, column, val, boost)
+
+Perform stochastic spawns to `w` from `column` with `val` walkers. `val * boost`
+controls the number of spawns performed.
+
+See [`SpawningStrategy`](@ref).
+"""
+@inline function spawn!(s::Exact, w, column, val, boost=1)
     T = valtype(w)
-    spawns = sum(offdiags; init=real(zero(T))) do (new_add, mat_elem)
+    attempts = 0
+    spawns = sum(offdiagonals(column); init=real(zero(T))) do (new_add, mat_elem)
         abs(projected_deposit!(
-            w, new_add, val * mat_elem, add => val, s.threshold
+            w, new_add, val * mat_elem, starting_address(column) => val, s.threshold
         ))
+        attempts += 1
     end
-    return (length(offdiags), spawns)
+    return (attempts, spawns)
```

Comment:
This looks wrong to me. While `attempts` will correctly count the attempts the result of `spawn` will not be correct. You'd need to increase the counter before computing the return value `abs(projected_deposit!(...))` within the `do ... end` construct.